x/oracle/client/cli/tx.go

```diff
@@ -0,0 +1,172 @@
+		Example: fmt.Sprintf(`%[1]s tx oracle update pb1skjwj5whet0lpe65qaq4rpq03hjxlwd9nf39lk --deposit 50000nhash`, version.AppName),
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			authority := authtypes.NewModuleAddress(govtypes.ModuleName)
+
+			msg := types.NewMsgUpdateOracle(
+				authority.String(),
+				args[0],
+			)
+
+			proposal, govErr := govcli.ReadGovPropFlags(clientCtx, cmd.Flags())
+			if govErr != nil {
+				return govErr
+			}
+			anys, govErr := sdktx.SetMsgs([]sdk.Msg{msg})
+			if govErr != nil {
+				return govErr
+			}
+			proposal.Messages = anys
+			var req sdk.Msg = proposal
+
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), req)
+		},
+	}
+
+	cmd.Flags().Bool(FlagGovProposal, false, "Run transaction as gov proposal")
+	govcli.AddGovPropFlagsToCmd(cmd)
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+// GetCmdSendQuery is a command to send a query to another chain's oracle
+func GetCmdSendQuery() *cobra.Command {
+	decoder := newArgDecoder(asciiDecodeString)
+	cmd := &cobra.Command{
+		Use:     "send-query <channel-id> <json>",
+		Short:   "Send a query to an oracle on a remote chain via IBC",
+		Args:    cobra.ExactArgs(2),
+		Aliases: []string{"sq"},
+		Example: fmt.Sprintf(`%[1]s tx oracle send-query channel-1 '{"query_version":{}}'`, version.AppName),
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			channelID := args[0]
+
+			queryData, err := decoder.DecodeString(args[1])
+			if err != nil {
+				return fmt.Errorf("decode query: %w", err)
+			}
+			if !json.Valid(queryData) {
+				return errors.New("query data must be json")
+			}
+
+			msg := types.NewMsgSendQueryOracle(
+				clientCtx.GetFromAddress().String(),
+				channelID,
+				queryData,
+			)
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+type argumentDecoder struct {
+	// dec is the default decoder
+	dec                func(string) ([]byte, error)
+	asciiF, hexF, b64F bool
+}
+
+func newArgDecoder(def func(string) ([]byte, error)) *argumentDecoder {
+	return &argumentDecoder{dec: def}
+}
+
+// DecodeString decodes the supplied json string
+func (a *argumentDecoder) DecodeString(s string) ([]byte, error) {
+	found := -1
+	for i, v := range []*bool{&a.asciiF, &a.hexF, &a.b64F} {
+		if !*v {
+			continue
+		}
+		if found != -1 {
+			return nil, errors.New("multiple decoding flags used")
+		}
+		found = i
+	}
+	switch found {
+	case 0:
+		return asciiDecodeString(s)
+	case 1:
+		return hex.DecodeString(s)
+	case 2:
+		return base64.StdEncoding.DecodeString(s)
+	default:
+		return a.dec(s)
+	}
+}
```

Comment:
I'm not sure what the purpose of the `asciiF`, `hexF`, and `b64F` bools is. They're never assigned to, so they're always false.

I feel like there's a missing flag on the command that would let the user tell this how to try to decode it. As it is, it'll only ever use `asciiDecodeString`.

Reply:
This was taken from wasmd, but I can simplify it for our use case.

Reply:
I think something is missing from that copy then, because those flags it loops through are never anything other than `false`.

Reply:
Ah! I think, maybe, there's some flag bindings that should be added to that command that bind a flag value to the fields in the decoder.  That would explain why the decoder is created outside of the command runner.

Reply:
I ended up removing this entirely and used the default decoder since we didn't have a use case for multiple types. Thanks for pointing this out.